synfig-studio/src/gui/widgets/widget_timetrack.cpp

```diff
@@ -199,6 +200,42 @@ bool Widget_Timetrack::move_selected(synfig::Time delta_time)
 	return ok;
 }
 
+ void Widget_Timetrack::interpolate_selected(synfig::Interpolation type)
+{
+	synfig::Waypoint::Model model;
+	model.set_before(type);
+	model.set_after(type);
+	std::vector<WaypointItem*> selection = waypoint_sd.get_selected_items();
+
+	synfigapp::Action::PassiveGrouper group(get_canvas_interface()->get_instance().get(),_("Change Waypoint Interpolation"));
+
+	for (WaypointItem* waypoint_item : selection) {
+		std::set<synfig::Waypoint, std::less<synfig::UniqueID> > waypoint_set_new;
+		fetch_waypoints(*waypoint_item, waypoint_set_new);
+
+		for (const synfig::Waypoint& waypoint_new : waypoint_set_new) {
+			synfig::Waypoint waypoint(waypoint_new);
+			if (waypoint.get_before() != type || waypoint.get_after() != type){
+				waypoint.apply_model(model);
```

Comment:
hmm, I'm not really ssure what you mean here.  I thought maybe you mean to set the waypoint parameter for multiple waypoints then perform the action once, but it gives out errors (also I think setting the parameter "waypoint" multiple time before perform action willl just overwrite it so its probably not what you meant)

Reply:
> hmm, I'm not really ssure what you mean here. I thought maybe you mean to set the waypoint parameter for multiple waypoints then perform the action once, but it gives out errors 

Yes.

> (also I think setting the parameter "waypoint" multiple time before perform action will just overwrite it so its probably not what you meant)

You can check `WaypointSet` action to be sure it doesn't - it adds it to the vector. 

The only issue I see here, is the `value_node` parameter. I don't know why it is passed, if it can be resolved from waypoint parameter. Also `value_node` overwrites previous value, so you need to check if:
1) it is used
2) it can't be resolved from waypoint parameter.

Reply:
> You can check `WaypointSet` action to be sure it doesn't - it adds it to the vector.

oh yes I checked it does add them to a vector but what is puzzling me is the value node, it is also what is causing the error when trying to perform the action for all the selected waypoints at once, the value node isn't set for each one of them. In fact I'm confused how it could be set for each one of them when there is only one value node handle in the `WayPointSet` ... maybe we need a value node vector similar to the waypoint vector or maybe there is one i'm not sure i will try to investigate here a bit.

> The only issue I see here, is the `value_node` parameter. I don't know why it is passed, if it can be resolved from waypoint parameter. Also `value_node` overwrites previous value, so you need to check if:
> 
> 1. it is used
> 2. it can't be resolved from waypoint parameter.

1- I believe yes it is used as it seems for the action `WaypointSet` to be performed it needs the waypoint and the valuenode of the waypoint, failure to have the valuenode gives out the error `unable to find waypoint` 

2- i'm not sure what you mean exactly by resolve, however the valuenode of the waypoint can be gotten from the waypoint itself using for example `get_parent_value_node`, im thinking maybe i can add an extra default parameter for doing waypoint related actions like this without having to set the valuenode explicitly for each one, to make it implicit by making the valuenode be set from `WaypointSet` itself and stored in a vector. But I have to first understand a bit more of whats going on, so I will have to investigate further.

Reply:
@ice0 , @rodolforg would this change be necessary here though, as in this PR. from what I see currently a similar situation which is setting interpolation from right mouse click groups the set waypoint actions into a passive group similar to what is done here, however to instead set the action once then waypointset class would have to be modified somewhat significantly.

Reply:
> would this change be necessary here though

Ok, this can be fixed later.